File: coursera/mmds-001/1.1.R
M = matrix(c(0,1/2,1/2,0,0,1,0,0,1),ncol = 3)
e = matrix(c(1,1,1),ncol = 1)
v1 = matrix(c(1,1,1),ncol = 1)
v1 = v1 / 3
n = 3
for (i in 1:5) {
    v1 = ((0.7 * M )  %*% v1) + (((1-0.7)*e)/n)
}

v1 = v1 * 3

File: coursera/mmds-001/1.1.py
import numpy

# M = numpy.matrix([[0, 0, 1],
                 #[1.0 / 2.0, 0, 0],
                 #[1.0 / 2.0, 1, 0]])
M = numpy.matrix([[0, 0, 0],
                  [1.0 / 2.0, 0, 0],
                  [1.0 / 2.0, 1.0, 1.0]])
e = numpy.matrix([[1.0 / 3.0], [1.0 / 3.0], [1.0 / 3.0]])
v = numpy.matrix([[1.0], [1.0], [1.0]])
print v

for i in range(50):
    v = 0.7 * M * v + (1 - 0.7) * e
    print 'iteration %d' % (i + 1), v
